contracts/service-registry/src/state.rs

```diff
@@ -130,16 +131,73 @@ pub enum AuthorizationState {
     Authorized,
 }
 
-// maps service_name -> Service
-pub const SERVICES: Map<&str, Service> = Map::new("services");
-// maps (service_name, chain_name, worker_address) -> ()
-pub const WORKERS_PER_CHAIN: Map<(&str, &ChainName, &Addr), ()> = Map::new("workers_per_chain");
-// maps (service_name, worker_address) -> Worker
-pub const WORKERS: Map<(&str, &Addr), Worker> = Map::new("workers");
+type ChainNames = HashSet<ChainName>;
+type ServiceName = str;
+type WorkerAddress = Addr;
+
+pub const SERVICES: Map<&ServiceName, Service> = Map::new("services");
+pub const WORKERS_PER_CHAIN: Map<(&ServiceName, &ChainName, &WorkerAddress), ()> =
+    Map::new("workers_per_chain");
+pub const CHAINS_PER_WORKER: Map<(&ServiceName, &WorkerAddress), ChainNames> =
+    Map::new("chains_per_worker");
+pub const WORKERS: Map<(&ServiceName, &WorkerAddress), Worker> = Map::new("workers");
+
+pub fn register_chain_support_utility(
+    storage: &mut dyn Storage,
+    service_name: String,
+    chains: Vec<ChainName>,
+    caller: WorkerAddress,
```

Comment:
```suggestion
    worker: WorkerAddress,
```
"caller" doesn't really mean anything to the state. "verifier" would be better, but I guess we keep it consistent for now and rename in bulk

Reply:
Good call!